include/seastar/http/internal/content_source.hh

```diff
@@ -40,15 +40,17 @@ namespace internal {
  * */
 class content_length_source_impl : public data_source_impl {
     input_stream<char>& _inp;
-    size_t _remaining_bytes = 0;
-    size_t* _consumed_bytes;
+    size_t _builtin_remaining_bytes = 0;
+    size_t& _remaining_bytes;
 public:
-    content_length_source_impl(input_stream<char>& inp, size_t length, size_t* consumed_bytes)
-        : _inp(inp), _remaining_bytes(length), _consumed_bytes(consumed_bytes) {
+    content_length_source_impl(input_stream<char>& inp, size_t length, size_t& remaining)
+        : _inp(inp), _remaining_bytes(remaining)
```

Comment:
I dont follow here, why you are first initialize the reference in member initialization list and the update the value of the `remaining` argument effectively overwriting whatever was stored there?